components/Dialogs/tests/snapshot/MDCAlertControllerDynamicTypeTests.m

```diff
@@ -0,0 +1,283 @@
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryExtraExtraExtraLarge {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategoryExtraExtraExtraLarge];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryAccessibilityMedium {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategoryAccessibilityMedium];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryAccessibilityLarge {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategoryAccessibilityLarge];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryAccessibilityExtraLarge {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategoryAccessibilityExtraLarge];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryAccessibilityExtraExtraLarge {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:
+            UIContentSizeCategoryAccessibilityExtraExtraLarge];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryAccessibilityExtraExtraExtraLarge {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:
+            UIContentSizeCategoryAccessibilityExtraExtraExtraLarge];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryExtraSmallAndLegacyDisabled {
```

Comment:
Is this test and the others like it misnamed? It looks like `adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable` is enabled, which is the legacy behavior.

Reply:
Fixed

Reply:
This still looks incorrect. Why is this method named `LegacyDisabled` if we're enabling the legacy behavior?

Reply:
Sorry I forgot to push